core/src/main/java/com/nerds/gamejam/ecs/system/RenderSystem.java

```diff
@@ -49,18 +45,21 @@ protected void begin() {
 
     @Override
     protected void process(int e) {
-        drawComposites(e);
-        drawSingleSprite(e);
-        drawFont(e);
-        drawAnimation(e);
-    }
+        PositionComponent position = positionMapper.get(e);
+        TextureReferenceComponent textureReference = textureReferenceMapper.get(e);
+        BodyComponent bodyComponent = bodyMapper.get(e);
+        ScaleComponent scaleComponent = scaleMapper.get(e);
 
-    private void drawSingleSprite(int e) {
-        SpriteComponent spriteComponent = this.spriteMapper.get(e);
-        if (spriteComponent != null) {
-            PositionComponent positionComponent = this.positionMapper.get(e);
-            drawSprite(spriteComponent.sprite, positionComponent);
-        }
+        float width = scaleComponent!= null ? bodyComponent.getWidth() * scaleComponent.getX() : bodyComponent.getWidth();
+        float height = scaleComponent != null ? bodyComponent.getHeight() * scaleComponent.getY() : bodyComponent.getHeight();
+
+        textureReference.getReferences().forEach(reference -> {
```

Comment:
Best to use a simple for loop here to remove the under the hood creation of iterator (less gc pressure minor as it is)

Reply:
tbh, I think it's even more negligible when the for i loop needs extra variables assigned. It's not super important either way, I just tend to lean towards readable code at that point

Reply:
It is very micro optimisation level but it is good practice to try and minimise object allocation. The variables would just be references versus an actual iterator creation plus the actual method calls and so on under the hood vs array indexing
Meh :-)